File: sdk/samples/FrameworkSamples/CFX/Serialization/Xml Serialization/SchemaImporterExtension/cs/orderschemaimporterextension/orderschemaimporterextension/orderschemaimporterextension.cs
#region Using directives

using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Text;
using System.Xml.Serialization;
using System.Xml.Serialization.Advanced;
using System.Xml.Schema;
using System.Xml;
using System.CodeDom;
using System.CodeDom.Compiler;
using Microsoft.Samples;

#endregion

namespace Microsoft.Samples
{
	public class OrderSchemaImporterExtension : SchemaImporterExtension
	{
		static OrderSchemaImporterExtension()
		{
		}

		public override string ImportSchemaType(XmlSchemaType type,
			XmlSchemaObject context, XmlSchemas schemas, XmlSchemaImporter importer,
			CodeCompileUnit compileUnit, CodeNamespace codeNamespace, 
            CodeGenerationOptions options, CodeDomProvider codeGenerator)
		{

			ConsoleColor defaultColor = Console.BackgroundColor;
			Console.BackgroundColor = ConsoleColor.Blue;
			Console.WriteLine("\tOrderExtension called for type: " + type.ToString());
			Console.BackgroundColor = defaultColor;

			return null;
		}

	
		public override string ImportSchemaType(string name, string ns, 
			XmlSchemaObject context, XmlSchemas schemas, XmlSchemaImporter importer, 
			CodeCompileUnit compileUnit, CodeNamespace codeNamespace, 
			CodeGenerationOptions options, CodeDomProvider codeGenerator)
		{
			ConsoleColor defaultColor = Console.BackgroundColor;
			Console.BackgroundColor = ConsoleColor.Blue;
			Console.WriteLine("\tOrderExtension called for name = " + name + " and namespace = " + ns);
			Console.BackgroundColor = defaultColor;

			if (name.Equals("Order") && ns.Equals("http://orders/"))
			{
				Console.BackgroundColor = ConsoleColor.Blue;
				Console.WriteLine();
				Console.WriteLine("\t\tOrderExtension matched for Order!");
				Console.WriteLine();
				Console.BackgroundColor = defaultColor;

				compileUnit.ReferencedAssemblies.Add("Order.dll");
				codeNamespace.Imports.Add(new CodeNamespaceImport("Microsoft.Samples"));
				return "Order";
			}


			return null;
		}
	}
}

File: sdk/samples/FrameworkSamples/CFX/Serialization/Xml Serialization/SchemaImporterExtension/vb/orderschemaimporterextension/orderschemaimporterextension/orderschemaimporterextension.vb
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.Collections.Specialized
Imports System.Text
Imports System.Xml.Serialization
Imports System.Xml.Serialization.Advanced
Imports System.Xml.Schema
Imports System.Xml
Imports System.CodeDom
Imports System.CodeDom.Compiler
Imports Microsoft.Samples.Orders


    Public Class OrderSchemaImporterExtension
    Inherits SchemaImporterExtension

    Shared Sub New()
    End Sub 'New


    Public Overloads Overrides Function ImportSchemaType(ByVal type As XmlSchemaType, _
            ByVal context As XmlSchemaObject, ByVal schemas As XmlSchemas, _
            ByVal importer As XmlSchemaImporter, ByVal compileUnit As CodeCompileUnit, _
            ByVal codeNamespace As CodeNamespace, ByVal options As CodeGenerationOptions, _
            ByVal codeGenerator As CodeDomProvider) As String

        Dim defaultColor As ConsoleColor = Console.BackgroundColor
        Console.BackgroundColor = ConsoleColor.Blue
        Console.WriteLine((vbTab & "OrderExtension called for type: " & type.ToString()))
        Console.BackgroundColor = defaultColor

        Return Nothing
    End Function 'ImportSchemaType



    Public Overloads Overrides Function ImportSchemaType(ByVal name As String, _
            ByVal ns As String, ByVal context As XmlSchemaObject, _
            ByVal schemas As XmlSchemas, ByVal importer As XmlSchemaImporter, _
            ByVal compileUnit As CodeCompileUnit, ByVal codeNamespace As CodeNamespace, _
            ByVal options As CodeGenerationOptions, ByVal codeGenerator As CodeDomProvider) As String
        Dim defaultColor As ConsoleColor = Console.BackgroundColor
        Console.BackgroundColor = ConsoleColor.Blue
        Console.WriteLine((ControlChars.Tab & "OrderExtension called for name = " & name & " and namespace = " & ns))
        Console.BackgroundColor = defaultColor

        If name.Equals("Order") And ns.Equals("http://orders/") Then
            Console.BackgroundColor = ConsoleColor.Blue
            Console.WriteLine()
            Console.WriteLine(vbTab & vbTab & "OrderExtension matched for Order!")
            Console.WriteLine()
            Console.BackgroundColor = defaultColor

            compileUnit.ReferencedAssemblies.Add("Order.dll")
            codeNamespace.Imports.Add(New CodeNamespaceImport("Microsoft.Samples.Orders"))
            Return "Order"
        End If


        Return Nothing
    End Function 'ImportSchemaType

End Class 'OrderSchemaImporterExtension
